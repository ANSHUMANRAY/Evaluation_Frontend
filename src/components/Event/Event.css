.event{
    height: fit-content;
    width: 300px;
    border-radius: 20px;
    margin: 10px;
    text-align: center;
    border: 1px solid #ccc;
    color: white;
}

.eventImg{
    width: 100%;
    height: 200px;
    border-radius: 20px 20px 0 0;
}

.eventStatus{
    width: 300px;
    height: 50px;
    display: flex;
    justify-content: space-around;
    align-items: center;
}

.bookmark{
    width: 20px;
}

.check{
    width: 20px;
}

.checkDiv{
    display: flex;
    color: #A0F3AD;
}

.checkDivSeat{
    display: flex;
    color: #EA8282;
}

.description{
    overflow: hidden;
   text-overflow: ellipsis;
   display: -webkit-box;
   -webkit-line-clamp: 3;
   -webkit-box-orient: vertical;
}

.venue{
    overflow: hidden;
   text-overflow: ellipsis;
   display: -webkit-box;
   -webkit-line-clamp: 1;
   -webkit-box-orient: vertical;
}