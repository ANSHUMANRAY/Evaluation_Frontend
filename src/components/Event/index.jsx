/* eslint-disable react/prop-types */
import React from 'react';
import axios from 'axios';
// import PropTypes from 'prop-types';
import check from '../../assets/circle-check-solid.svg';
import cross from '../../assets/circle-xmark-solid.svg';
import bookmarked from '../../assets/bookmark-solid.svg';
import bookmark from '../../assets/bookmark-regular.svg';
import './Event.css';
import { ThemeContext } from '../../contexts/themeContext';
import { useNavigate } from 'react-router-dom';

export default function Event(props) {
  const navigate = useNavigate();
  const {theme} = React.useContext(ThemeContext);
  const { event, isChange } = props;
  const [isBookmarked, setIsBookmarked] = React.useState(event.isBookmarked);
  const bookmarkIcon = isBookmarked ? bookmarked : bookmark;

  const handleBookmark = () => {
    axios.patch(`http://localhost:8000/api/events/${event.id}`, {isBookmarked: !isBookmarked})
      .then(() => {
        setIsBookmarked(!isBookmarked);
        isChange();
      });
  };

  const handleClick = () => {
    navigate(`/events/${event.id}`);
  };
  return (
    <div style={{backgroundColor:`${theme}`}} className="event">
      <img onClick={handleClick} src={event.imgUrl} alt={event.name} className="eventImg" />
      <div className="eventInfo">
        <p className='eventName'>{event.name}</p>
        <p className='description'>{event.description}</p>
        <p className='venue'>VENUE:{event.venue}</p>
        <p >DATE:{event.datetime}</p>
      </div>
      <div className="eventStatus">
        <div>
          {event.isRegistered ? (
            <div className='checkDiv'>
              <img className='check' src={check}/>
              <p>Registered</p>
            </div>
          ) : (event.areSeatsAvailable ? (
            <div className='checkDivSeat'>
              <img className='check' src={cross}/>
              <p>No Seats Available</p>
            </div>
          ) : (<div></div>))}
        </div>
        <div onClick={handleBookmark}>
          <img className='bookmark' src={bookmarkIcon} alt='bookmark' />
        </div>
      </div>
    </div>
  );
}

// Event.propTypes = {
//   event: PropTypes.object().isRequired,
// };