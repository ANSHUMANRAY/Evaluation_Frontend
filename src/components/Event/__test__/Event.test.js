import React from 'react';
import {render} from '@testing-library/react';
import Event from '../index';
import { ThemeContext } from '../../../contexts/themeContext';
import { mockEvent } from '../../../mocks/mockEvents';

const mockNavigate= jest.fn();
jest.mock('react-router-dom', () => ({
  useNavigate: () => mockNavigate,
}));

describe('Event', () => {
  it('should render correctly', () => {
    const {container} = render(<ThemeContext.Provider value={{theme: 'black', setTheme: jest.fn()}}><Event event={mockEvent}/></ThemeContext.Provider>);
    expect(container).toMatchSnapshot();
  });
});