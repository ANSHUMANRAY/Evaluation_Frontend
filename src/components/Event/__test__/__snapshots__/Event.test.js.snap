// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event should render correctly 1`] = `
<div>
  <div
    class="event"
    style="background-color: black;"
  >
    <img
      alt="Battle of the Bands"
      class="eventImg"
      src="https://i.ibb.co/3zbdvWX/battle-of-bands.jpg"
    />
    <div
      class="eventInfo"
    >
      <p
        class="eventName"
      >
        Battle of the Bands
      </p>
      <p
        class="description"
      >
        Get ready for Battle of the Bands, where the hottest up-and-coming rock groups will compete for the ultimate prize. With heart-pumping beats and electrifying performances, you won't want to miss this adrenaline-fueled event. Each band will bring their A-game, leaving everything on the stage for the chance to be crowned champion. Don't miss out on the most unforgettable rock competition of the year!
      </p>
      <p
        class="venue"
      >
        VENUE:
        All Stars Arena, Las Vegas, NV, USA
      </p>
      <p>
        DATE:
        2023-03-01T05:00:00.000Z
      </p>
    </div>
    <div
      class="eventStatus"
    >
      <div>
        <div
          class="checkDivSeat"
        >
          <img
            class="check"
            src="circle-xmark-solid.svg"
          />
          <p>
            No Seats Available
          </p>
        </div>
      </div>
      <div>
        <img
          alt="bookmark"
          class="bookmark"
          src="bookmark-regular.svg"
        />
      </div>
    </div>
  </div>
</div>
`;
